Chaincase.Common/Contracts/ITorManager.cs

```diff
@@ -17,5 +17,9 @@ public interface ITorManager
         Task StopAsync();
 
         Task StartAsync(bool ensureRunning, string dataDir);
+
+        string CreateHiddenServiceAsync();
+
+        void DestroyHiddenServiceAsync(string serviceId);
```

Comment:
We didn't think much about managing hidden service identifying keys on first pass. `DestroyHiddenServiceAsync` removes the ephemeral service if I remember correctly.